x/evm/keeper/evm.go

```diff
@@ -29,6 +30,23 @@ func (k *Keeper) HandleInternalEVMCall(ctx sdk.Context, req *types.MsgInternalEV
 	return &sdk.Result{Data: ret}, nil
 }
 
+func (k *Keeper) HandleInternalEVMDelegateCall(ctx sdk.Context, req *types.MsgInternalEVMDelegateCall) (*sdk.Result, error) {
+	if !k.IsCWCodeHashWhitelistedForEVMDelegateCall(ctx, req.CodeHash) {
+		return nil, errors.New("code hash not authorized to make EVM delegate call")
+	}
+	var to *common.Address
+	if req.To != "" {
```

Comment:
are there cases where req.To == "" ?

Reply:
in such case `to` should be left as `nil`